scripts/wallet.js

```diff
@@ -1073,23 +1030,87 @@ export class Wallet {
     }
 
     /**
-     * Finalize Transaction. To be called after it's signed and sent to the network, if successful
-     * @param {Transaction} transaction
+     * Adds a transaction to the mempool. To be called after it's signed and sent to the network, if successful
+     * @param {import('./transaction.js').Transaction} transaction
      */
-    finalizeTransaction(transaction) {
+    async addTransaction(transaction, skipDatabase = false) {
+        this.#mempool.addTransaction(transaction);
+        let i = 0;
+        for (const out of transaction.vout) {
+            this.updateHighestUsedIndex(out);
+            const status = this.getScriptType(out.script);
+            if (status & OutpointState.OURS) {
+                this.#mempool.setOutpointStatus(
+                    new COutpoint({
+                        txid: transaction.txid,
+                        n: i,
+                    }),
+                    status
+                );
+            }
+            i++;
+        }
+
         if (transaction.hasShieldData) {
             wallet.#shield.finalizeTransaction(transaction.txid);
         }
 
-        mempool.updateMempool(transaction);
-        mempool.setBalance();
+        if (skipDatabase) {
```

Comment:
shouldn't it be  `if (!skipDatabase) {`? or I am not understanding the meaning of `skipDatabase`?